### PR TITLE
docs: upgrade guide refactor organizes releases into their own pages; moves upgrade guide to the root

### PR DESCRIPTION
**Pull request by chri2547**
This PR organizes Grafana releases into their own upgrade pages; moves the upgrade guide to the root.

### CODE REVIEW

File: docs/sources/upgrade-guide/upgrade-v8.0/index.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  - documentation
+  - upgrade
+title: Upgrade to Grafana v8.0
+menutitle: Upgrade to v8.0
+weight: 2900
+---
+
+# Upgrade to Grafana v8.0
+
+{{< docs/shared "upgrade/upgrade-common-tasks.md" >}}
```
Comment (jdbaldry) [inline docs/sources/upgrade-guide/upgrade-v8.0/index.md:17]: I'd recommend using the updated `docs/shared` parameter syntax described in https://grafana.com/docs/writers-toolkit/writing-guide/shortcodes/#docsshared-shortcode.

The updated parameter syntax provides better feedback for user errors like referencing a file that does not exist.

You can also offset the headings using the shortcode if you need using the `leveloffset` parameter.

```suggestion
{{< docs/shared lookup="upgrade/upgrade-common-tasks.md" source="grafana" >}}
```

Same for all other instances.

File: docs/sources/upgrade-guide/upgrade-v8.0/index.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
+---
+aliases:
```
Comment (jdbaldry) [inline docs/sources/upgrade-guide/upgrade-v8.0/index.md:2]: I think we should remove the empty aliases line.
```suggestion
```

Same for other upgrade guides.

File: docs/sources/upgrade-guide/upgrade-v8.0/index.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
```
Comment (jdbaldry) [inline docs/sources/upgrade-guide/upgrade-v8.0/index.md:3]: Remove empty line
```suggestion
```

Same for other upgrade guides.

File: docs/sources/upgrade-guide/upgrade-v9.3/index.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
+---
+aliases:
+  - ../installation/upgrading/
```
Comment (jdbaldry) [inline docs/sources/upgrade-guide/upgrade-v9.3/index.md:3]: I think the second redirect is necessary to redirect the now removed page `setup-grafana/upgrade-grafana/`.

```suggestion
  - ../installation/upgrading/
  - ../setup-grafana/upgrade-grafana/
```

File: docs/sources/shared/upgrade/upgrade-common-tasks.md [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+---
+
+We recommend that you upgrade Grafana often to stay current with the latest fixes and enhancements.
+Because Grafana upgrades are backward compatible, the upgrade process is straightforward, and dashboards and graphs will not change.
+
+In addition to common tasks you should complete for all versions of Grafana, there might be additional upgrade tasks to complete for a version.
+
+> **Note:** There might be minor breaking changes in some releases. We outline these changes in the [What's New ]({{< relref "../whatsnew/" >}}) document for each release.
+
+For versions of Grafana prior to v9.2, we published additional information in the [Release Notes]({{< relref "../release-notes/" >}}).
```
Comment (jdbaldry) [inline docs/sources/shared/upgrade/upgrade-common-tasks.md:12]: Hugo reports this as a broken relref.

```suggestion
For versions of Grafana prior to v9.2, we published additional information in the [Release Notes]({{< relref "../../release-notes/" >}}).
```

File: docs/sources/shared/upgrade/upgrade-common-tasks.md [hunk 6]
```diff
@@ -0,0 +1,181 @@
+---
+title: Upgrade guide common tasks
+---
+
+We recommend that you upgrade Grafana often to stay current with the latest fixes and enhancements.
+Because Grafana upgrades are backward compatible, the upgrade process is straightforward, and dashboards and graphs will not change.
+
+In addition to common tasks you should complete for all versions of Grafana, there might be additional upgrade tasks to complete for a version.
+
+> **Note:** There might be minor breaking changes in some releases. We outline these changes in the [What's New ]({{< relref "../whatsnew/" >}}) document for each release.
```
Comment (jdbaldry) [inline docs/sources/shared/upgrade/upgrade-common-tasks.md:10]: Hugo reports this as a broken relref.

```suggestion
> **Note:** There might be minor breaking changes in some releases. We outline these changes in the [What's New ]({{< relref "../../whatsnew/" >}}) document for each release.
```

File: docs/sources/shared/upgrade/upgrade-common-tasks.md [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+
+To upgrade Grafana running on Centos or RHEL, complete the following steps:
+
+1. In your current installation of Grafana, save your custom configuration changes to a file named `<grafana_install_dir>/conf/custom.ini`.
+
+   This enables you to upgrade Grafana without the risk of losing your configuration changes.
+
+1. Perform one of the following steps based on your installation.
+
+   - If you [downloaded an RPM package](https://grafana.com/grafana/download) to install Grafana, then complete the steps documented in [Install on RPM-based Linux]({{< relref "./installation/rpm" >}}) to upgrade Grafana.
```
Comment (jdbaldry) [inline docs/sources/shared/upgrade/upgrade-common-tasks.md:118]: Hugo reports this as a broken relref.

```suggestion
   - If you [downloaded an RPM package](https://grafana.com/grafana/download) to install Grafana, then complete the steps documented in [Install on RPM-based Linux]({{< relref "../../setup-grafana/installation/rpm/" >}}) to upgrade Grafana.
```

**Review by jdbaldry — CHANGES_REQUESTED** [review]
I've found some broken links that will need fixing:

```
WARN 2023/01/30 11:53:35 [en] REF_NOT_FOUND: Ref "../setup-grafana/upgrade-grafana/": "/hugo/content/docs/grafana/next/release-notes/release-notes-7-3-0.md:59:110": page not found
WARN 2023/01/30 11:53:35 [en] REF_NOT_FOUND: Ref "../../upgrade-grafana/": "/hugo/content/docs/grafana/next/setup-grafana/installation/debian/index.md:22:157": page not found
WARN 2023/01/30 11:53:35 [en] REF_NOT_FOUND: Ref "../../upgrade-grafana/": "/hugo/content/docs/grafana/next/setup-grafana/installation/mac/index.md:13:211": page not found
WARN 2023/01/30 11:53:35 [en] REF_NOT_FOUND: Ref "../../upgrade-grafana/": "/hugo/content/docs/grafana/next/setup-grafana/installation/rpm/index.md:23:157": page not found
WARN 2023/01/30 11:53:35 [en] REF_NOT_FOUND: Ref "../../upgrade-grafana/": "/hugo/content/docs/grafana/next/setup-grafana/installation/windows/index.md:13:26": page not found
WARN 2023/01/30 11:53:35 [en] REF_NOT_FOUND: Ref "../setup-grafana/upgrade-grafana/": "/hugo/content/docs/grafana/next/whatsnew/whats-new-in-v7-1.md:114:21": page not found
WARN 2023/01/30 11:53:35 [en] REF_NOT_FOUND: Ref "../setup-grafana/upgrade-grafana/": "/hugo/content/docs/grafana/next/whatsnew/whats-new-in-v7-2.md:120:33": page not found
WARN 2023/01/30 11:53:35 [en] REF_NOT_FOUND: Ref "../setup-grafana/upgrade-grafana/": "/hugo/content/docs/grafana/next/whatsnew/whats-new-in-v7-0.md:231:21": page not found
WARN 2023/01/30 11:53:35 [en] REF_NOT_FOUND: Ref "../setup-grafana/upgrade-grafana/": "/hugo/content/docs/grafana/next/whatsnew/whats-new-in-v7-2.md:150:21": page not found
WARN 2023/01/30 11:53:35 [en] REF_NOT_FOUND: Ref "../setup-grafana/upgrade-grafana/": "/hugo/content/docs/grafana/next/whatsnew/whats-new-in-v7-3.md:152:21": page not found
WARN 2023/01/30 11:53:35 [en] REF_NOT_FOUND: Ref "../setup-grafana/upgrade-grafana/": "/hugo/content/docs/grafana/next/whatsnew/whats-new-in-v7-4.md:243:21": page not found
```

File: docs/sources/upgrade-guide/upgrade-v8.0/index.md (hunk 3 above)
Comment (chri2547) [reply docs/sources/upgrade-guide/upgrade-v8.0/index.md:3]: 👍

**Review by tleopold — APPROVED** [review]
I didn't have a chance to review this in great detail but it looks good to me.  Definitely a great improvement over what was there before!

**Review by jdbaldry — APPROVED** [review]
Links after the reorganization are work as expected. I have commented nits for style in the shared content.

File: docs/sources/upgrade-guide/upgrade-v8.1/index.md [hunk 8]
```diff
@@ -0,0 +1,29 @@
+---
+aliases:
```
[the remark this answers is not in the document]
Comment (chri2547) [reply docs/sources/upgrade-guide/upgrade-v8.1/index.md:2]: I've addressed them in another PR.

**Pull request merged**